xikolo-ios/Views/CourseItemCell.swift

```diff
@@ -75,16 +75,23 @@ class CourseItemCell : UITableViewCell {
     }
 
     func configure(for courseItem: CourseItem, with configuration: CourseItemCellConfiguration) {
+        let isAvailable = !configuration.inOfflineMode || (courseItem.content?.isAvailableOffline ?? false)
+
+        self.isUserInteractionEnabled = isAvailable
+
         self.item = courseItem
         self.titleView.text = courseItem.title
-        self.titleView.alpha = (configuration.inOfflineMode && !(courseItem.content?.isAvailableOffline ?? false)) ? 0.5 : 1.0
+        self.titleView.textColor = isAvailable ? UIColor.black : UIColor.lightGray
+        self.detailLabel.textColor = isAvailable ? UIColor.darkText : UIColor.lightGray
 
+        self.iconView.tintColor = isAvailable ? UIColor.black : UIColor.lightGray
         if let iconName = courseItem.iconName {
-            self.iconView.image = UIImage(named: "item-\(iconName)-28")
+            self.iconView.image = UIImage(named: "item-\(iconName)-28")?.withRenderingMode(.alwaysTemplate)
         }
 
         let wasVisitedBefore = courseItem.visited ?? true
-        self.readStateView.backgroundColor = wasVisitedBefore ? UIColor.clear : Brand.TintColor
+        self.readStateView.alpha = wasVisitedBefore ? 0.0 : 1.0
+        self.readStateView.backgroundColor = isAvailable ? Brand.TintColor : UIColor.lightGray
```

Comment:
This is probably the more readable and shorter code version but it could make sense to properly configure the different view states and switch them then. The disabled gray color could then also be set in the storyboard and wouldn't need to appear in code as a "magic color".

Reply:
You need ro refer to the 'disabled color' in code because tableview cells are getting reused. So you can't assume this color being set via the storyboard. But I agree that the current cell configuration could be improved. Unfortunately, the cell's state can't be represented by a simple enum due to the different states for offline/online, download and preloading. So I would handle this in a separate issue if required.

Reply:
True, I forgot about that. Sounds good :)

Reply:
But I think you'd still not need to use colours here, just set the correct state again for the reused cell